source/Calamari/ArgoCD/Git/RepositoryWrapper.cs

```diff
@@ -72,56 +83,92 @@ public void StageFiles(string[] filesToStage)
+                                                           new GitBranchName(currentBranchName),
+                                                           cancellationToken);
             }
         }
 
         string CalculateBranchName()
         {
             return $"octopus-argo-cd-{Guid.NewGuid().ToString("N").Substring(0, 10)}";
         }
-        
+
         public void PushChanges(string branchName)
         {
-            Remote remote = repository.Network.Remotes["origin"];
-            repository.Branches.Update(repository.Head, 
+            var remote = repository.Network.Remotes["origin"];
+            repository.Branches.Update(repository.Head,
                                        branch => branch.Remote = remote.Name,
                                        branch => branch.UpstreamBranch = $"refs/heads/{branchName}");
-            
+
             PushStatusError? errorsDetected = null;
             var pushOptions = new PushOptions
             {
                 CredentialsProvider = (url, usernameFromUrl, types) =>
                                           new UsernamePasswordCredentials { Username = connection.Username, Password = connection.Password },
                 OnPushStatusError = errors => errorsDetected = errors
             };
-            
+
             repository.Network.Push(repository.Head, pushOptions);
             if (errorsDetected != null)
             {
                 throw new CommandException($"Failed to push to branch {branchName} - {errorsDetected.Message}");
             }
         }
-        
-        string GenerateCommitMessage(string summary, string description)
+
+        static string GenerateCommitMessage(string summary, string description)
         {
             return description.Equals(string.Empty)
                 ? summary
                 : $"{summary}\n\n{description}";
         }
+
+        public void Dispose()
+        {
+            //free up the repository handles
+            repository?.Dispose();
+
+            //delete the repository local
+            log.Verbose("Attempting to delete cloned repository local directory");
+            try
+            {
+                //some files in the .git folder can/are ReadOnly which makes them impossible to delete
+                //so just remove the ReadOnly attribute from all files (if they are ReadOnly)
+                foreach (var gitFile in calamariFileSystem.EnumerateFilesWithGlob(repoCheckoutDirectoryPath, ".git/**/*"))
+                {
+                    calamariFileSystem.RemoveReadOnlyAttributeFromFile(gitFile);
+                }
+
+                calamariFileSystem.DeleteDirectory(repoCheckoutDirectoryPath);
+                log.Verbose("Deleted cloned repository local directory");
+            }
+            catch (Exception e)
+            {
+                log.VerboseFormat("Failed to delete cloned repository local directory.{0}{1}", Environment.NewLine, e);
```

Comment:
If this is exceptional, probably log it at a higher level

Reply:
My reasoning is that it's not something the user can fix and when the task completes, it'll be cleaned up anyway.